src/main/java/com/firebolt/jdbc/connection/FireboltConnectionServiceSecret.java

```diff
@@ -100,15 +106,24 @@ private FireboltProperties getSessionPropertiesForSystemEngine(String accessToke
         String systemEngineEndpoint = fireboltGatewayUrlService.getUrl(accessToken, accountName);
         FireboltAccount account = fireboltAccountIdService.getValue(accessToken, accountName);
         infraVersion = account.getInfraVersion();
+        URL systemEngienUrl = UrlUtil.createUrl(systemEngineEndpoint);
+        Map<String, String> systemEngineUrlUrlParams = UrlUtil.getQueryParameters(systemEngienUrl);
+        String accountId = systemEngineUrlUrlParams.getOrDefault(ACCOUNT_ID.getKey(), account.getId());
```

Comment:
I think the idea here is to use all parameters and not only `account_id`. Yes, currently (and maybe even further) we only receive `account_id`, but generally this assumes the ability to receive other parameters.
So generally, we should take all received parameters (not only `account_id`) and add them when we connect to a system engine.

Reply:
Also, if account_id is not provided in parameters, we should not provide it either instead of getting it manually and setting it as default.
For example, we can receive a URL like `https:/01hnj9r1xrx3a4t3kb1ec7qs2b.api.us-east-1.staging.firebolt.io`, where account_id is included in the endpoint itself, so there is no need to add a query parameter as well

Reply:
Ok, as for parameter makes sense, didn't see that.
As for account_id, my point is that we need to forcefully unset it (even if it was provided by a user). As I understand from the code, account_id value can still exist in the url even if it wasn't provided in system engine url response. Let me know if I'm missing someting

Reply:
This logic is implemented in StatemtentClientImpl: lines 310-331

Reply:
Ok, makes sense